homeassistant/components/intellifire/number.py

```diff
@@ -0,0 +1,74 @@
+"""Flame height number sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.number import (
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, LOGGER
+from .coordinator import IntellifireDataUpdateCoordinator
+from .entity import IntellifireEntity
+
+
+@dataclass
+class IntellifireFlameControlEntity(IntellifireEntity, NumberEntity):
+    """Flame height control entity."""
+
+    _attr_native_max_value: float = 5
+    _attr_native_min_value: float = 1
+    _attr_native_step: float = 1
+    _attr_mode: NumberMode = NumberMode.SLIDER
+
+    def __init__(
+        self,
+        coordinator: IntellifireDataUpdateCoordinator,
+        description: NumberEntityDescription,
+    ) -> None:
+        """Initilaize Flame height Sensor."""
+        super().__init__(coordinator, description)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the current Flame Height segment number value."""
+        value = self.coordinator.read_api.data.flameheight + 1
+        LOGGER.debug("%s Value: %s", self._attr_name, value)
```

Comment:
Is this needed, even at debug-level? This will log a statement every time the value is read.